--- clive-0.4.11/src/clive/parse.py~	2008-04-27 23:11:12.000000000 +0300
+++ clive-0.4.11/src/clive/parse.py	2008-04-27 23:11:27.000000000 +0300
@@ -52,6 +52,7 @@
             ('metacafe.', 'metac', self._parse_metacafe),
             ('delfi.', 'delfi', self._parse_delfi),
             ('www.southparkzone.com', 'southparkzone', self._parse_southparkzone),
+            ('www.reporter.ee', 'reporter', self._parse_reporter),
         ]
 
     ## Parses a video page data (HTML)
@@ -297,6 +298,12 @@
         vid = md5.new(str(time.time())).hexdigest()[:8]
         return (url, vid, low_quality)
 
+    def _parse_reporter(self, url, data, low_quality):
+        # http://www.reporter.ee/index.php/2008/04/21/soiduopetaja-vorpis-poiste-nupeldamisest-videoklippe/
+        url = self._parse_from_to(data, 'so.addVariable("src", "', '"', skip_from = 1);
+        vid = md5.new(str(time.time())).hexdigest()[:8]
+        return (url, vid, low_quality)
+
     def _random_vid(self):
         return md5.new(str(time.time())).hexdigest()[:8]
 
