--- clive-0.4.1/clive/parse.py	2008-01-06 02:40:04.000000000 +0200
+++ clive-0.4.1-delfi/clive/parse.py	2008-01-20 00:31:04.282874939 +0200
@@ -49,6 +49,7 @@
 			('dailymotion.', 'dmotion', self._parse_dmotion),
 			('guba.com', 'guba', self._parse_guba),
 			('stage6.', 'stage6', self._parse_stage6),
+			('delfi.', 'delfi', self._parse_delfi),
 		]
 
 	## Parses a video page data (HTML).
@@ -184,6 +185,16 @@
 
 		return (url, vid)
 
+	def _parse_delfi(self, url, data):
+		text = self._parse_from_to(data, "flv_url:", ",")
+		try:
+			url = urllib.unquote(text.replace("'", "").split('flv_url: ',2)[1])
+		except IndexError:
+			pass
+
+		vid = md5.new(str(time.time())).hexdigest()[:8]
+		return (url, vid)
+
 	def _parse_from_to(self, data, _from, to, skip_from=0):
 		start = data.find(_from)
 		end = data.find(to, start)
