--- clive-2.1.5/clive~  2009-02-10 11:09:45.000000000 +0100
+++ clive-2.1.5/clive   2009-02-10 11:13:10.000000000 +0100
@@ -86,6 +86,7 @@
     IsLiveleak  => qr|liveleak.com|i,
     IsEvisor    => qr|evisor.tv|i,
     IsDelfi     => qr|.delfi.|i,
+    IsReporter  => qr|www.reporter.ee|i,
 );

 # Parse config
@@ -306,6 +307,8 @@
         ($xurl, $id) = handle_liveleak($response_ref, $response_fh);
     } elsif ( $url =~ /$re_hosts{IsDelfi}/ ) {
         ($xurl, $id) = handle_delfi($response_ref, $response_fh);
+    } elsif ( $url =~ /$re_hosts{IsReporter}/ ) {
+        ($xurl, $id) = handle_reporter($response_ref, $response_fh);
     }
 #    elsif ( $url =~ /$re_hosts{IsMetacafe}/ ) {
 #        ($xurl, $id) = handle_metacafe($response_ref);
@@ -369,6 +372,14 @@
                         $content_ok = 1;
                     }
                 }
+            # Reporter returns "flv-application/octet-stream"
+            } elsif ( $content_type =~ m!application/octet-stream! ) {
+                if ( $entry{page_url} =~ /$re_hosts{IsReporter}/ ) {
+                    if ( $opts{format} eq "flv" ) {
+                        $entry{file_suffix} = 'flv';
+                        $content_ok = 1;
+                    }
+                }
             }
             $errmsg = "expected different content-type, "
                 . "received \"$content_type\"" unless $content_ok;
@@ -773,6 +784,27 @@
     return ($xurl, $id);
 }
 
+sub handle_reporter {
+    my ($response_ref, $response_fh) = @_;
+
+    my %re = (
+    # http://www.reporter.ee/index.php/2008/04/21/soiduopetaja-vorpis-poiste-nupeldamisest-videoklippe/
+        GrabVideoPrefix => qr|obj\s+=\s+{type:"video",file:"(\w+://[^"]+)|,
+        GrabVideoID     => qr|attributes\s+=\s+{id:\s+"video([^"]+)"|,
+    );
+
+    my $xurl;
+    my $id     = $1 if $$response_ref =~ /$re{GrabVideoID}/;
+    my $prefix = $1 if $$response_ref =~ /$re{GrabVideoPrefix}/;
+
+    if ($id and $prefix) {
+        $xurl = $prefix . $id . '.flv';
+    } else {
+        print STDERR "error: url not found\n";
+    }
+
+    return ($xurl, $id);
+}
 
 # Subroutines: Progress
 # NOTE: the 'dot' progress copies much from wget.
