--- clive-2.1.10/clive~ 2009-04-27 11:29:24.000000000 +0200
+++ clive-2.1.10/clive  2009-04-27 11:30:52.000000000 +0200
@@ -88,6 +88,7 @@
                  IsDmotion   => qr|dailymotion.com|i,
                  IsCctv      => qr|tv.cctv.com|i,
                  IsDelfi     => qr|.delfi.|i,
+                IsReporter  => qr|www.reporter.ee|i,
                );

 my @re_hosts_arr = (
@@ -101,6 +102,7 @@
                     [$re_hosts{IsDmotion},   \&handle_dmotion],
                     [$re_hosts{IsCctv},      \&handle_cctv],
                     [$re_hosts{IsDelfi},     \&handle_delfi],
+                   [$re_hosts{IsReporter},  \&handle_reporter],
                    );

 # Parse config
@@ -369,6 +372,14 @@
                         $content_ok = 1;
                     }
                 }
+            # Reporter returns "flv-application/octet-stream"
+            } elsif ( $content_type =~ m!application/octet-stream! ) {
+                if ( $entry{page_url} =~ /$re_hosts{IsReporter}/ ) {
+                    if ( $opts{format} eq "flv" ) {
+                        $entry{file_suffix} = 'flv';
+                        $content_ok = 1;
+                    }
+                }
             }
             $errmsg = "expected different content-type, "
                 . "received \"$content_type\"" unless $content_ok;
@@ -1061,6 +1062,28 @@
     return ($xurl, $id);
 }
 
+sub handle_reporter {
+    my ($response_ref, $response_fh) = @_;
+
+    my %re = (
+    # http://www.reporter.ee/index.php/2008/04/21/soiduopetaja-vorpis-poiste-nupeldamisest-videoklippe/
+        GrabVideoPrefix => qr|obj\s+=\s+{type:"video",file:"(\w+://[^"]+)|,
+        GrabVideoID     => qr|attributes\s+=\s+{id:\s+"video([^"]+)"|,
+    );
+
+    my $xurl;
+    my $id     = $1 if $$response_ref =~ /$re{GrabVideoID}/;
+    my $prefix = $1 if $$response_ref =~ /$re{GrabVideoPrefix}/;
+
+    if ($id and $prefix) {
+        $xurl = $prefix . $id . '.flv';
+    } else {
+        print STDERR "error: url not found\n";
+    }
+
+    return ($xurl, $id);
+}
+
 # Subroutines: Progress
 # NOTE: the 'dot' progress copies much from wget.
 
