--- clive-2.1.9/clive~  2009-03-30 11:55:42.000000000 +0200
+++ clive-2.1.9/clive   2009-03-30 12:05:00.000000000 +0200
@@ -85,6 +85,7 @@
                  IsLiveleak  => qr|liveleak.com|i,
                  IsEvisor    => qr|evisor.tv|i,
                  IsDmotion   => qr|dailymotion.com|i,
+                 IsDelfi     => qr|.delfi.|i,
                );

 # Parse config
@@ -324,6 +325,8 @@
         ($xurl, $id) = handle_evisor($response_ref);
     } elsif ($url =~ /$re_hosts{IsDmotion}/) {
         ($xurl, $id) = handle_dmotion($response_ref);
+    } elsif ( $url =~ /$re_hosts{IsDelfi}/ ) {
+       ($xurl, $id) = handle_delfi($response_ref, $response_fh);
     }
     return -1
         if !$xurl or !$id or !$title;
@@ -992,6 +993,64 @@
     return ($xurl, $id);
 }
 
+sub handle_delfi {
+    my ($response_ref, $response_fh) = @_;
+ 
+    my %re = (
+    # videobox
+    # http://www.delfi.ee/news/paevauudised/paevavideo/article.php?id=15218215
+        GrabVideoboxURL => qr|flv_url:\s*'(.*?)'|,
+    # videoproject embed
+    # http://www.delfi.ee/news/paevauudised/paevavideo/article.php?id=18759038
+        GrabVideoSalt   => qr|_delfiVideoSalt\s*=\s*"([^"]+)";|,
+        GrabVideoSite   => qr|src="(\S+://[^/]+)/js/embed.js"|,
+    # videoproject
+    # http://video.delfi.ee/video/CzurzqNz/
+        GrabVideoURL   => qr|\.addVariable\('file',\s*'([^']+)'|,
+    );
+
+    my ($xurl, $id);
+    my $videobox_url = $1 if $$response_ref =~ /$re{GrabVideoboxURL}/;
+    my $video_salt   = $1 if $$response_ref =~ /$re{GrabVideoSalt}/;
+    my $video_site   = $1 if $$response_ref =~ /$re{GrabVideoSite}/;
+    my $video_url   = $1 if $$response_ref =~ /$re{GrabVideoURL}/;
+
+    if ($videobox_url) {
+        $id = $1 if $videobox_url =~ m{/([^/]+)\.flv$};
+        $xurl = $videobox_url;
+
+    } elsif ($video_url) {
+        $xurl = uri_unescape($video_url);
+        $id = $1 if $xurl =~ m{^\S+://[^/]+/v/(.+?)\.flv};
+
+    } elsif ($video_salt and $video_site) {
+        my $url = $video_site . '/video/' . $video_salt . '/';
+        print "done.\nfetch page ..." unless $opts{quiet};
+
+        my $page = "";
+        open my $fh, ">", \$page;
+
+        # Disable: header
+        $curl->setopt(CURLOPT_HEADER, 0);
+        $curl->setopt(CURLOPT_URL, $url);
+        $curl->setopt(CURLOPT_WRITEDATA, $fh);
+
+        my $rc = $curl->perform;
+        close $fh;
+
+        if ($rc == 0) {
+            # recurse
+            ($xurl, $id) = handle_delfi(\$page);
+        } else {
+            print STDERR "\nerror: " . $curl->strerror($rc) . " (http/$rc)\n";
+        }
+    } else {
+        print STDERR "error: url not found\n";
+    }
+
+    return ($xurl, $id);
+}
+
 # Subroutines: Progress
 # NOTE: the 'dot' progress copies much from wget.
