--- clive-0.4.10/src/clive/parse.py~	2008-04-25 23:53:39.000000000 +0300
+++ clive-0.4.10/src/clive/parse.py	2008-04-25 23:55:53.000000000 +0300
@@ -50,6 +50,7 @@
             ('dailymotion.', 'dmotion', self._parse_dmotion),
             ('guba.com', 'guba', self._parse_guba),
             ('metacafe.', 'metac', self._parse_metacafe),
+			('delfi.', 'delfi', self._parse_delfi),
         ]
 
     ## Parses a video page data (HTML)
@@ -221,6 +222,16 @@
         url = self._parse_from_to(data, 'mediaURL=', '&', skip_from=1)
         return (url, vid, low_quality)
 
+	def _parse_delfi(self, url, data, low_quality):
+		text = self._parse_from_to(data, "flv_url:", ",")
+		try:
+			url = urllib.unquote(text.replace("'", "").split('flv_url: ',2)[1])
+		except IndexError:
+			pass
+
+		vid = md5.new(str(time.time())).hexdigest()[:8]
+		return (url, vid, low_quality)
+
     def _random_vid(self):
         return md5.new(str(time.time())).hexdigest()[:8]
 
